File: is_prime.py
import math


def is_prime(n):
  prime_boolean = True
  for m in range(2, math.floor(n/2)):
    if n % m == 0:
      prime_boolean = False

  return prime_boolean

print("testing is_prime on input 59...")
assert is_prime(59) is True, "is_prime(59) should equal True"
print("PASSED")

print("testing is_prime on input 51...")
assert is_prime(51) is False, "is_prime(51) should equal False"
print("PASSED")

File: isPrime.cpp
#include <iostream>
#include <cassert>

bool isPrime(int x)
{
  bool prime_boolean = true;
  if(x == 0 || x ==1){
    return false;
  };
  for(int n = 2; n <= x/2; n++){
    if((x%n) == 0){
      prime_boolean = false;
    };
  };
   
  return prime_boolean;
}

int main()
{
  assert(!isPrime(0));
  assert(!isPrime(1));
  assert(isPrime(2));
  assert(isPrime(3));
  assert(!isPrime(4));
  assert(isPrime(5));
  assert(isPrime(7));
  assert(!isPrime(9));
  assert(isPrime(11));
  assert(isPrime(13));
  assert(!isPrime(15));
  assert(!isPrime(16));
  assert(isPrime(17));
  assert(isPrime(19));
  assert(isPrime(97));
  assert(!isPrime(99));
  assert(!isPrime(99));
  assert(isPrime(13417));

  std::cout << "Success!";

  return 0;
}